src/table/components/head/HeadCellContent.tsx

```diff
@@ -1,41 +1,72 @@
-import React from "react";
+import React, { useRef, useState } from "react";
 
+import HeadCellMenu, { MenuAvailabilityFlags } from "@qlik/nebula-table-utils/lib/components/HeadCellMenu";
+import { SortDirection } from "../../../types";
 import { TableContext, useContextSelector } from "../../context";
 import { HeadCellContentProps } from "../../types";
-import { areTabStopsEnabled } from "../../utils/accessibility-utils";
+import { areTabStopsEnabled, setFocusOnClosetColumnAdjuster } from "../../utils/accessibility-utils";
 import getHeadIcons from "../../utils/get-head-icons";
-import HeadCellMenu from "./HeadCellMenu";
-import { StyledHeadCellContent, StyledHeadCellIconWrapper, StyledSortButton, VisuallyHidden } from "./styles";
+import { handleHeadCellMenuKeyDown } from "../../utils/handle-keyboard";
+import OldHeadCellMenu from "./HeadCellMenu";
+import {
+  AbsolutelyStyledRefAnchor,
+  StyledHeadCellContent,
+  StyledHeadCellIconWrapper,
+  StyledSortButton,
+  VisuallyHidden,
+} from "./styles";
 
-const HeadCellContent = ({ children, column, isActive, isInteractionEnabled }: HeadCellContentProps) => {
-  const { keyboard, translator, changeSortOrder } = useContextSelector(TableContext, (value) => value.baseProps);
+const HeadCellContent = ({ children, column, isSorted, isInteractionEnabled }: HeadCellContentProps) => {
+  const { keyboard, translator, changeSortOrder, interactions, embed, app, model } = useContextSelector(
+    TableContext,
+    (value) => value.baseProps
+  );
   const isFocusInHead = useContextSelector(TableContext, (value) => value.focusedCellCoord[0] === 0);
+  const isNewHeadCellMenuEnabled = useContextSelector(
+    TableContext,
+    (value) => value.featureFlags.isNewHeadCellMenuEnabled
+  );
+
+  const [open, setOpen] = useState(false);
+  const anchorRef = useRef<HTMLDivElement>(null);
+  const listboxRef = useRef<HTMLDivElement>(null);
 
   const { startIcon, endIcon, lockIcon } = getHeadIcons(column);
   const tabIndex = isInteractionEnabled && areTabStopsEnabled(keyboard) ? 0 : -1;
 
-  const handleSort = () => isInteractionEnabled && changeSortOrder(column);
+  const handleSort = () => isInteractionEnabled && changeSortOrder(column, column.sortDirection);
```

Comment:
why are you adding an argument here? I thought with the flag off, you could still use `changeSortOrder` as before